apps/heft/src/plugins/TypeScriptPlugin/TypeScriptBuilder.ts

```diff
@@ -794,15 +805,135 @@ export class TypeScriptBuilder extends SubprocessRunnerBase<ITypeScriptBuilderCo
     return diagnostic.category;
   }
 
+  private _getProgramWithPatchedEmit<Program extends TTypescript.BuilderProgram | TTypescript.Program>(
+    ts: ExtendedTypeScript,
+    program: Program
+  ): Program {
+    if ((program as Program & { __heftPatched?: boolean }).__heftPatched) {
+      return program;
+    }
+    (program as Program & { __heftPatched?: boolean }).__heftPatched = true;
+
+    const originalEmit: Program['emit'] = program.emit;
+    program.emit = (...args) => {
```

Comment:
I seem to recall trying to call emit() multiple times originally, and running into an issue that it doesn't work when invoked on a BuilderProgram, because calling `emit()` has side effects (namely that it clears the list of files that are pending emit).

Reply:
Likely this needs to be careful to modify the inner program and not the builder program instance.

Reply:
I was also worried about inner/outer here. All of the tests pass, so, unless this is one of those untested cases, I'm not totally sure what the deal is. If you think there's a test within the repo I can run that would test this, I'd appreicate some instructions on how to do this. I've just been running `rush test` which I've assumed does the right thing.

Reply:
rush test will clean files, and `rush-lib` uses multi-emit for commonJS for Jest and ESM for webpack, so that should suffice for the non-incremental non-watch case. I have a local variant on this change to experiment with that also cleans up some other quirks to bring in improvements I did during the upcoming Heft refactor (namely passing a custom function when creating a watch host, solution host, etc. that wraps ts.createEmitAndSemanticDiagnosticsBuilderProgram instead of passing it directly).

Reply:
> uses multi-emit for commonJS for Jest and ESM for webpack, so that should suffice for the non-incremental non-watch case.

Ah, okay. That's the test that failed until I finished off the change, so, it seems like it's good.